16/umbraco-cms/customizing/foundation/fetching-data/fetch-api.md

```diff
@@ -0,0 +1,179 @@
+```typescript
+import { UMB_AUTH_CONTEXT } from '@umbraco-cms/backoffice/auth';
+import type { UmbClassInterface } from '@umbraco-cms/backoffice/class-api';
+
+/**
+ * Make an authorized request to any Backoffice API.
+ * @param host A reference to the host element that can request a context.
+ * @param url The URL to request.
+ * @param method The HTTP method to use.
+ * @param body The body to send with the request (if any).
+ * @returns The response from the request as JSON.
+ */
+async function makeRequest(host: UmbClassInterface, url: string, method = 'GET', body?: any) {
+  const authContext = await host.getContext(UMB_AUTH_CONTEXT);
+  const token = await authContext?.getLatestToken();
+  const response = await fetch(url, {
+    method,
+    body: body ? JSON.stringify(body) : undefined,
+    headers: {
+      'Authorization': `Bearer ${token}`,
+      'Content-Type': 'application/json',
+    },
+  });
+  return response.json();
+}
+```
+
+The above example serves to illustrate some of the process to make a request to the Management API. You can use this function to make requests to any endpoint in the Management API. The function does not handle errors or responses, so you will need to add that logic yourself, nor does it handle the authentication process. If the token is timed out, you will get a 401 error back, if the `getLatestToken` method failed to refresh the token.
+
+## Executing the request
+
+Regardless of method, you can execute the fetch requests through Umbraco's [tryExecute](https://apidocs.umbraco.com/v16/ui-api/classes/packages_core_auth.UmbAuthContext.html#tryexecute) function. This function will handle any errors that occur during the request and will automatically refresh the token if it is expired. If the session is expired, the function will also make sure the user logs in again.
+
+**Example:**
+
+```javascript
+import { tryExecute } from '@umbraco-cms/backoffice/resources';
+
+const request = makeRequest(this, '/umbraco/management/api/v1/server/status');
+const response = await tryExecute(this, request);
+
+if (response.error) {
+    console.error('There was a problem with the fetch operation:', response.error);
+} else {
+    console.log(response); // Do something with the data
+}
+```
+
+You can read more about the `tryExecute` function in this article:
+
+{% content-ref url="try-execute.md" %}
+[try-execute.md](try-execute.md)
+{% endcontent-ref %}
+
+## Conclusion
+
+The Fetch API is a powerful and flexible way to make network requests in JavaScript. It is available in all modern browsers and is the recommended way to make network requests in JavaScript. The Fetch API can be used in Umbraco to make network requests to the server. It can also be used to make requests to the Management API controllers. You can use the Fetch API to make requests to any endpoint in the Management API. You can also use it to handle responses in a variety of formats. This is especially useful, if you have but a few requests to make.
```

Comment:
```suggestion
The Fetch API is a powerful and flexible way to make network requests in JavaScript. It is available in all modern browsers and is the recommended way to make network requests in JavaScript. The Fetch API can be used in Umbraco to make network requests to the server. It can also be used to make requests to the Management API controllers. You can use the Fetch API to make requests to any endpoint in the Management API. You can also use it to handle responses in a variety of formats. This is useful if you only need to make a few requests.
```